provider/provider_yaml_test.go

```diff
@@ -459,7 +468,431 @@ func TestS3BucketObjectDeprecation(t *testing.T) {
+		// by Terraform, but they have different requirements for successful tag
+		// interactions. That is why we have tests for both resources.
+		{
+			name: "bucket", token: "aws:s3:BucketV2", typ: "aws:s3/bucketV2:BucketV2",
+			tags: map[string]interface{}{
+				"LocalTag": "foo",
+			},
+			postUpHook: func(t *testing.T, outputs auto.OutputMap) {
+				validateOutputTags(outputs, map[string]interface{}{
+					"LocalTag":  "foo",
+					"GlobalTag": "bar",
+				})
+			},
+			defaultTags: map[string]interface{}{
+				"GlobalTag": "bar",
+			},
+		},
+		{
+			name: "bucket_ignore_tags", token: "aws:s3:BucketV2", typ: "aws:s3/bucketV2:BucketV2",
+			tags: map[string]interface{}{
+				"LocalTag": "foo",
+			},
+			postUpHook: func(t *testing.T, outputs auto.OutputMap) {
+				validateOutputTags(outputs, map[string]interface{}{
+					"LocalTag":  "foo",
+					"GlobalTag": "bar",
+				})
+			},
+			defaultTags: map[string]interface{}{
+				"GlobalTag": "bar",
+			},
+			ignoreTagKeys: []string{"IgnoreKey"},
+			preImportHook: func(t *testing.T, outputs auto.OutputMap) {
+				t.Helper()
+				resArn := outputs["id"].Value.(string)
+				addBucketTags(context.Background(), resArn, map[string]string{
+					"IgnoreKey": "foo",
+				})
+			},
+		},
+		{
+			name: "sdkv2", token: "aws:cognito:UserPool", typ: "aws:cognito/userPool:UserPool",
+			tags: map[string]interface{}{
+				"LocalTag": "foo",
+			},
+			defaultTags: map[string]interface{}{
+				"GlobalTag": "bar",
+			},
+			postUpHook: func(t *testing.T, outputs auto.OutputMap) {
+				validateOutputTags(outputs, map[string]interface{}{
+					"LocalTag":  "foo",
+					"GlobalTag": "bar",
+				})
+			},
+			properties: map[string]interface{}{
+				// aliasAttributes is necessary because otherwise we don't
+				// see a clean initial refresh
+				"aliasAttributes": []interface{}{"email"},
+			},
+		},
+
+		// A PF resource (appconfig:Environment)
+		// PF resources deal with tags differently
+		{
+			name: "pf", token: "aws:appconfig:Environment", typ: "aws:appconfig/environment:Environment",
+			tags: map[string]interface{}{
+				"LocalTag": "foo",
+			},
+			defaultTags: map[string]interface{}{
+				"GlobalTag": "bar",
+			},
+			postUpHook: func(t *testing.T, outputs auto.OutputMap) {
+				validateOutputTags(outputs, map[string]interface{}{
+					"LocalTag":  "foo",
+					"GlobalTag": "bar",
+				})
+			},
+			other: `
+  app:
+    type: aws:appconfig:Application`,
+			properties: map[string]interface{}{
+				"applicationId": "${app.id}",
+			},
+		},
+		{
+			name: "pf_ignore_tags", token: "aws:appconfig:Environment", typ: "aws:appconfig/environment:Environment",
+			tags: map[string]interface{}{
+				"LocalTag": "foo",
+			},
+			ignoreTagKeys: []string{"IgnoreKey"},
+			preImportHook: func(t *testing.T, outputs auto.OutputMap) {
+				t.Helper()
+				resArn := outputs["resArn"].Value.(string)
+				addAppconfigEnvironmentTags(context.Background(), resArn, map[string]string{
```

Comment:
Nit: we could generalize this using `tag:TagResources` which can tag all taggable resources.

Reply:
Updated!